tests/test_api.rs

```diff
@@ -4326,3 +4327,68 @@ fn clear_kept_objects() {
   scope.clear_kept_objects();
   eval(scope, step2).unwrap();
 }
+
+#[test]
+fn wasm_streaming_callback() {
+  thread_local! {
+    static WS: RefCell<Option<v8::WasmStreaming>> = RefCell::new(None);
+  }
+
+  let callback = |scope: &mut v8::HandleScope,
+                  url: v8::Local<v8::Value>,
+                  ws: v8::WasmStreaming| {
+    assert_eq!("https://example.com", url.to_rust_string_lossy(scope));
+    WS.with(|slot| assert!(slot.borrow_mut().replace(ws).is_none()));
+  };
+
+  let _setup_guard = setup();
+
+  let isolate = &mut v8::Isolate::new(v8::CreateParams::default());
+  isolate.set_wasm_streaming_callback(callback);
+
+  let scope = &mut v8::HandleScope::new(isolate);
+  let context = v8::Context::new(scope);
+  let scope = &mut v8::ContextScope::new(scope, context);
+
+  let script = r#"
+    globalThis.result = null;
+    WebAssembly
+      .compileStreaming("https://example.com")
+      .then(result => globalThis.result = result);
+  "#;
+  eval(scope, script).unwrap();
+
+  let global = context.global(scope);
+  let name = v8::String::new(scope, "result").unwrap().into();
+  assert!(global.get(scope, name).unwrap().is_null());
+
+  let mut ws = WS.with(|slot| slot.borrow_mut().take().unwrap());
+  assert!(global.get(scope, name).unwrap().is_null());
+
+  ws.on_bytes_received(&[0x00, 0x61, 0x73, 0x6d, 0x01, 0x00, 0x00, 0x00]);
```

Comment:
What is this wasm code - maybe put a comment with the WAT?

Reply:
It doesn't contain any actual WASM bytecode, just the file magic and WASM version. :-) I'll add a comment.